source/server/api_listener_impl.h

```diff
@@ -0,0 +1,176 @@
+      void addFilter(Network::FilterSharedPtr) override {}
+      void addReadFilter(Network::ReadFilterSharedPtr) override {}
+      bool initializeReadFilters() override { return true; }
+
+      // Network::Connection
+      void addConnectionCallbacks(Network::ConnectionCallbacks&) override {}
+      void addBytesSentCallback(Network::Connection::BytesSentCb) override {}
+      void enableHalfClose(bool) override {}
+      void close(Network::ConnectionCloseType) override {}
+      Event::Dispatcher& dispatcher() override { return parent_.parent_.dispatcher(); }
+      uint64_t id() const override { return 12345; }
+      std::string nextProtocol() const override { return EMPTY_STRING; }
+      void noDelay(bool) override {}
+      void readDisable(bool) override {}
+      void detectEarlyCloseWhenReadDisabled(bool) override {}
+      bool readEnabled() const override { return true; }
+      const Network::Address::InstanceConstSharedPtr& remoteAddress() const override {
+        return parent_.parent_.address();
+      }
+      absl::optional<Network::Connection::UnixDomainSocketPeerCredentials>
+      unixSocketPeerCredentials() const override {
+        return absl::nullopt;
+      }
+      const Network::Address::InstanceConstSharedPtr& localAddress() const override {
+        return parent_.parent_.address();
+      }
+      void setConnectionStats(const Network::Connection::ConnectionStats&) override {}
+      Ssl::ConnectionInfoConstSharedPtr ssl() const override { return nullptr; }
+      absl::string_view requestedServerName() const override { return EMPTY_STRING; }
+      State state() const override { return Network::Connection::State::Open; }
+      void write(Buffer::Instance&, bool) override {}
+      void setBufferLimits(uint32_t) override {}
+      uint32_t bufferLimit() const override { return 65000; }
+      bool localAddressRestored() const override { return false; }
+      bool aboveHighWatermark() const override { return false; }
+      const Network::ConnectionSocket::OptionsSharedPtr& socketOptions() const override {
+        return options_;
+      }
+      StreamInfo::StreamInfo& streamInfo() override { return stream_info_; }
+      const StreamInfo::StreamInfo& streamInfo() const override { return stream_info_; }
+      void setDelayedCloseTimeout(std::chrono::milliseconds) override {}
+      absl::string_view transportFailureReason() const override { return EMPTY_STRING; }
+
+      SyntheticReadCallbacks& parent_;
+      StreamInfo::StreamInfoImpl stream_info_;
+      Network::ConnectionSocket::OptionsSharedPtr options_;
+    };
+
+    HttpApiListenerImpl& parent_;
+    SyntheticConnection connection_;
+  };
+
+  const envoy::config::listener::v3alpha::Listener& config_;
+  ListenerManagerImpl& parent_;
+  const std::string name_;
+  Network::Address::InstanceConstSharedPtr address_;
+  ProtobufMessage::ValidationVisitor& validation_visitor_;
+  Stats::ScopePtr global_scope_;
```

Comment:
Why is this needed? Can't the global scope be used directly? Or is this to match the existing listeners?

Reply:
yep, I was just matching the existing listener here.